src/imp/platform/windows.rs

```diff
@@ -38,78 +34,35 @@ pub fn get_app_dir(t: AppDataType) -> Result<PathBuf, AppDirsError> {
     get_folder_path(folder_id).map(|os_str| os_str.into())
 }
 
-/// https://msdn.microsoft.com/en-us/library/dd378457.aspx#FOLDERID_RoamingAppData
-#[allow(non_upper_case_globals)]
-static FOLDERID_RoamingAppData: GUID = GUID {
-    Data1: 0x3EB685DB,
-    Data2: 0x65F9,
-    Data3: 0x4CF6,
-    Data4: [0xA0, 0x3A, 0xE3, 0xEF, 0x65, 0x72, 0x9F, 0x3D],
-};
-
-/// https://msdn.microsoft.com/en-us/library/dd378457.aspx#FOLDERID_LocalAppData
-#[allow(non_upper_case_globals)]
-static FOLDERID_LocalAppData: GUID = GUID {
-    Data1: 0xF1B32785,
-    Data2: 0x6FBA,
-    Data3: 0x4FCF,
-    Data4: [0x9D, 0x55, 0x7B, 0x8E, 0x7F, 0x15, 0x70, 0x91],
-};
-
-/// https://msdn.microsoft.com/en-us/library/dd378457.aspx#FOLDERID_ProgramData
-#[allow(non_upper_case_globals)]
-static FOLDERID_ProgramData: GUID = GUID {
-    Data1: 0x62AB5D82,
-    Data2: 0xFDC1,
-    Data3: 0x4DC3,
-    Data4: [0xA9, 0xDD, 0x07, 0x0D, 0x1D, 0x49, 0x5D, 0x97],
-};
-
-/// Wrapper around `winapi::PWSTR` to automatically free the string pointer.
+/// Wrapper around `windows::core::PWSTR` to automatically free the string pointer.
 /// This ensures the memory is freed when `get_folder_path` scope is left,
 /// regardless of whether the call succeeded or failed/panicked.
 struct SafePwstr(PWSTR);
 impl Drop for SafePwstr {
     fn drop(&mut self) {
-        unsafe { CoTaskMemFree(self.0 as *mut _) }
+        unsafe { CoTaskMemFree(Some(self.0.as_ptr() as *mut _)) }
     }
 }
 
 fn get_folder_path(folder_id: &GUID) -> Result<OsString, AppDirsError> {
     unsafe {
-        // Wide C string to be allocated by SHGetKnownFolderPath.
-        // We are responsible for freeing this!
-        let mut raw_path: PWSTR = ptr::null_mut();
-
         // SHGetKnownFolderPath arguments:
         // 1. reference to KNOWNFOLDERID
         // 2. no flags
-        // 3. null handle -> current user
-        // 4. output location
-        let result = SHGetKnownFolderPath(folder_id, 0, ptr::null_mut(), &mut raw_path);
-
-        // SHGetKnownFolderPath shouldn't ever fail, but if it does,
-        // it will return a negative HRESULT.
-        if result < 0 {
-            return Err(AppDirsError::NotSupported);
-        }
-
-        // Ensures that the PWSTR is free when we leave this scope through
-        // normal execution or a thread panic.
-        let _cleanup = SafePwstr(raw_path);
-
-        // Manually calculate length of wide C string.
-        let mut length = 0;
-        for i in 0.. {
-            if *raw_path.offset(i) == 0 {
-                length = i as usize;
-                break;
+        // 3. `None` handle -> current user
+        //
+        // Returns a PWSTR, which contains the path to requested folder.
+        match SHGetKnownFolderPath(folder_id, KNOWN_FOLDER_FLAG(0), None) {
```

Comment:
Is there a `_NONE` variant? Otherwise:


```suggestion
        match SHGetKnownFolderPath(folder_id, KNOWN_FOLDER_FLAG::default(), None) {
```